src/prompts.py

```diff
@@ -40,19 +40,45 @@
     Answer a question by first performing the most relevant search on arXiv and
     reading the abstracts of the papers found.
 
-    If the some (or all) the abstracts respond to the question, return the answer.
-    Otherwise, select only the most promising papers from the list access
-    and their whole content to search for the answer within.
+    **Limit searches to a maximum of 3 distinct search queries.**
+    **Limit article access to a maximum of 2 articles.**
 
-    Quote the papers you used to answer.
+    **Prioritize providing a concise overview of the answer using abstracts.**
+    **Only access articles if the abstracts are insufficient to answer the question, or to provide specific data.**
 
-    If you don't find the answer, say you did not find relevant information
-    and terminate your generation.
+    If you notice a pattern of repeated or very similar search queries, stop the search and provide an introductory answer based on the abstracts you have read.
+
+    If the some (or all) the abstracts respond to the question *comprehensively*, return the answer in the following JSON format:
+    {
+        "response": "The answer to the question.",
+        "article_list": ["url1", "url2", ...],
+        "source": "abstracts"
+    }
+
+    Otherwise, if the abstracts provide *some* information, and you need to access full articles, select only the **single most promising** paper (or at most two in very specific cases) from the list and use its content to complete the answer. Return the answer in the following JSON format:
+    {
+        "response": "The answer to the question.",
+        "article_list": ["url1", "url2", ...],
+        "source": "articles"
+    }
+
+    If you find that the search space is too broad or the question leads to an endless search for comparisons and details, provide a concise introductory answer and indicate that the topic has many facets.
+
+    **Always include the URLs of the abstracts used in the `article_list` when `source` is "abstracts".**
+    Quote the papers you used to answer, adding their urls to the article_list.
+
+    If you don't find the answer, say you did not find relevant information and terminate your generation,
+    returning the following JSON format:
+    {
```

Comment:
ah quindi gli devi dire esattamente come rispondere in questo case? Mi chiedo se basterebbe usare i defaults nel modello Pydantic invece?